src/main/java/aquality/selenium/core/elements/RelativeElementFinder.java

```diff
@@ -0,0 +1,52 @@
+package aquality.selenium.core.elements;
+
+import aquality.selenium.core.localization.ILocalizedLogger;
+import aquality.selenium.core.waitings.IConditionalWait;
+import org.openqa.selenium.By;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+/**
+ * Implementation of IElementFinder for a relative SearchContext.
+ */
+public class RelativeElementFinder extends ElementFinder {
+    private final IConditionalWait conditionalWait;
+    private final Supplier<SearchContext> searchContextSupplier;
+
+    public RelativeElementFinder(ILocalizedLogger localizedLogger, IConditionalWait conditionalWait,
+                                 Supplier<SearchContext> searchContextSupplier) {
+        super(localizedLogger, conditionalWait);
+        this.conditionalWait = conditionalWait;
+        this.searchContextSupplier = searchContextSupplier;
+    }
+
+    @Override
+    public List<WebElement> findElements(By locator, DesiredState desiredState, Long timeoutInSeconds) {
```

Comment:
if  I am right the difference between this method and another one from the ElementFinder class that here we use 'searchContextSupplier.get()' instead 'driver' - can we create one parametrized and re-use it for both cases by passing necessary supplier?

Reply:
actually no, you are wrong.
Nevertheless, I guess that I can extract the method from the lambda expression

Reply:
refactored, please check again